src/app/pages/wallets/my-wallets/pages-wallets-my-wallets.component.ts

```diff
@@ -37,4 +43,38 @@ export class PagesWalletsMyWalletsComponent implements OnInit {
       },
     });
   }
-}
+
+  public addNewWallet(userInputData: string) {
+    this.myWalletsService.addNewWallet(userInputData).subscribe( (wallet: MyWallet) => {
+        this.myWalletsData.data = [wallet, ...this.myWalletsData.data!]
+    });
+  }
+
+  public updateWallet(userInputData: string) {
+    this.myWalletsService.updateNewWallet(this.nowEditWallet!, userInputData).subscribe( (walletResponse: MyWallet) => {
+        this.myWalletsData.data = this.myWalletsData?.data?.map(wallet => {
+            if (wallet.id === this.nowEditWallet) {
+                return walletResponse;
+            }
+            return wallet;
+        }) as MyWallet[];
```

Comment:
You should not use `as X` if you really don't need. It is as bad as using `any`. I think the problem is because you have used `this.myWalletsData?.data?`. When editing wallet all data should be loaded so `this.myWalletsData.data` won't be `null` but `MyWallet[]`. Because of that you can "tell" Typescript that `this.myWallets.data` is not null by using `!`. So in the end it should look like: `this.myWalletsData.data!.map(` :)